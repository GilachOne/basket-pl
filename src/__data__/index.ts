import oopss from './animations/oopss.json';
import error from './animations/error.json';

export const animations = {
    oopss,
    error,
};
