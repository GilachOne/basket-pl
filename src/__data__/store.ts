import { configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/query';

import { api } from './api/main';
import { authApi } from './api/auth';
import { reducer as errorTextReducer } from './slices/errorTextSlice';

export const store = configureStore({
    reducer: {
        [api.reducerPath]: api.reducer,
        [authApi.reducerPath]: authApi.reducer,

        errorText: errorTextReducer,
    },

    middleware: getDefaultMiddleware =>
        getDefaultMiddleware()
            .concat(api.middleware)
            .concat(authApi.middleware),
});

setupListeners(store.dispatch);

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
