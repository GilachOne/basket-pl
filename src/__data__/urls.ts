import { getNavigationsValue } from '@ijl/cli';

const baseURL = getNavigationsValue('basket.main');

export const URLs = {
    landing: {
        url: `${baseURL}`,
        isEnabled: getNavigationsValue('basket.main'),
    },
    dashboard: {
        url: `${baseURL}${getNavigationsValue('basket.dashboard')}`,
        isEnabled: getNavigationsValue('basket.dashboard'),
    },
    signIn: {
        url: `${baseURL}${getNavigationsValue('basket.sign.in')}`,
        getUrlWidthEmail: email => {
            if (email) {
                return `${baseURL}${getNavigationsValue(
                    'basket.sign.in'
                )}?email=${email}`;
            }
            return `${baseURL}${getNavigationsValue('basket.sign.in')}`;
        },
        isEnabled: getNavigationsValue('basket.sign.in'),
    },
    signUp: {
        url: `${baseURL}${getNavigationsValue('basket.sign.up')}`,
        getUrlWidthEmail: email => {
            if (email) {
                return `${baseURL}${getNavigationsValue(
                    'basket.sign.up'
                )}?email=${email}`;
            }
            return `${baseURL}${getNavigationsValue('basket.sign.up')}`;
        },
        isEnabled: getNavigationsValue('basket.sign.up'),
    },
    list: {
        url: `${baseURL}${getNavigationsValue('basket.list')}`,
        getUrl: listId =>
            `${baseURL}${getNavigationsValue('basket.list')}`.replace(
                ':id',
                listId
            ),
        isEnabled: getNavigationsValue('basket.list'),
    },
    add: {
        url: `${baseURL}${getNavigationsValue('basket.add')}`,
        getUrl: listId =>
            `${baseURL}${getNavigationsValue('basket.add')}`.replace(
                ':id',
                listId
            ),
        isEnabled: getNavigationsValue('basket.add'),
    },
    resetPassword: {
        url: `${baseURL}${getNavigationsValue('basket.reset.password')}`,
        isEnabled: getNavigationsValue('basket.reset.password'),
    },
};
