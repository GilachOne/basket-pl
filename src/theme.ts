export const theme = {
    colors: {
        brand: {
            light2: '#08DA36',
            light: '#74D14C',
            main: '#74D14C',
            dark: '#54AB26',
        },
        background: {
            light: '#FFFFFF',
			main: '#008060',
        },
        accent: {
            light6: '#CCCCCC',
            light5: '#F6EAFF',
            light4: '#d2d8e0',
            light3: '#FAF9F8',
            light2: '#D2D5D9',
            light: '#F6EAFF',
            main: '#9828F0',
            dark: '#2C107A',
        },
        text: {
            light3: '#A68FE5',
            light2: '#42474C',
            light: '#6B7177',
            main: '#FFFFFF',
            dark: '#000000',
        },
        bg: {
            main: '#000000',
        },
        error: {
            main: '#FF0000',
        },
    },
};
