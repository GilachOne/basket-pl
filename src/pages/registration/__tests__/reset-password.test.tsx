import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { describe, it, expect } from '@jest/globals';
import { render, fireEvent, screen } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import 'whatwg-fetch';
import { Provider } from 'react-redux';

import ResetPassword from '../reset-password';
import { theme } from '../../../theme';
import { store } from '../../../__data__/store';

describe('testing ResetPassword render', () => {
    it('ResetPassword render', () => {
        const { container } = render(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <BrowserRouter>
                        <ResetPassword />
                    </BrowserRouter>
                </ThemeProvider>
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });

    it('Test clicks ResetPassword ', () => {
        const { container } = render(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <BrowserRouter>
                        <ResetPassword />
                    </BrowserRouter>
                </ThemeProvider>
            </Provider>
        );
        fireEvent.click(screen.getByRole('link', {
            name: /вход/i
          }));
        expect(container).toMatchSnapshot();
        fireEvent.click(screen.getByRole('link', {
            name: /регистрация/i
          }));
        expect(container).toMatchSnapshot();
          fireEvent.change(
            screen.getByPlaceholderText('Введите логин или почту'),
            { target: { value: 'email@test.ru' } }
        );
        fireEvent.change(
            screen.getByPlaceholderText('Введите проверочный код'),
            { target: { value: '12345"Dd' } }
        );
        fireEvent.change(
            screen.getByPlaceholderText('Введите новый пароль'),
            { target: { value: '12345"Dd' } }
            
        );
        fireEvent.click(screen.getByRole('button', {
            name: /сбросить/i
          }));
        expect(container).toMatchSnapshot();
     });
    
});
