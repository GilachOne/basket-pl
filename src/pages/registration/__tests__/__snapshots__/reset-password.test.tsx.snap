// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ResetPassword render ResetPassword render 1`] = `
<div>
  <form
    class="css-1jmybar"
  >
    <img
      class="css-1kpxy3o"
      src="logo.png"
    />
    <h1
      class="css-9jhoxj"
    >
      Сбросить пароль
    </h1>
    <div
      class="css-1bej84q"
    >
      <div
        class="css-bpveru"
      >
        <a
          class="css-som9qd"
          href="/url/url/"
        >
          Вход
        </a>
      </div>
      <div
        class="css-bw50an"
      >
        <a
          class="css-som9qd"
          href="/url/url/"
        >
          Регистрация
        </a>
      </div>
    </div>
    <input
      class="css-1wjmxg1"
      name="email"
      placeholder="Введите логин или почту"
      required=""
      type="text"
    />
    <input
      class="css-1wjmxg1"
      name="code"
      placeholder="Введите проверочный код"
      required=""
      type="text"
    />
    <input
      class="css-1wjmxg1"
      id="password-input"
      name="password"
      placeholder="Введите новый пароль"
      required=""
      type="password"
    />
    <button
      class="css-19ic31z"
      type="submit"
    >
      Сбросить
    </button>
  </form>
</div>
`;

exports[`testing ResetPassword render Test clicks ResetPassword  1`] = `
<div>
  <form
    class="css-1jmybar"
  >
    <img
      class="css-1kpxy3o"
      src="logo.png"
    />
    <h1
      class="css-9jhoxj"
    >
      Сбросить пароль
    </h1>
    <div
      class="css-1bej84q"
    >
      <div
        class="css-bpveru"
      >
        <a
          class="css-som9qd"
          href="/url/url/"
        >
          Вход
        </a>
      </div>
      <div
        class="css-bw50an"
      >
        <a
          class="css-som9qd"
          href="/url/url/"
        >
          Регистрация
        </a>
      </div>
    </div>
    <input
      class="css-1wjmxg1"
      name="email"
      placeholder="Введите логин или почту"
      required=""
      type="text"
    />
    <input
      class="css-1wjmxg1"
      name="code"
      placeholder="Введите проверочный код"
      required=""
      type="text"
    />
    <input
      class="css-1wjmxg1"
      id="password-input"
      name="password"
      placeholder="Введите новый пароль"
      required=""
      type="password"
    />
    <button
      class="css-19ic31z"
      type="submit"
    >
      Сбросить
    </button>
  </form>
</div>
`;

exports[`testing ResetPassword render Test clicks ResetPassword  2`] = `
<div>
  <form
    class="css-1jmybar"
  >
    <img
      class="css-1kpxy3o"
      src="logo.png"
    />
    <h1
      class="css-9jhoxj"
    >
      Сбросить пароль
    </h1>
    <div
      class="css-1bej84q"
    >
      <div
        class="css-bpveru"
      >
        <a
          class="css-som9qd"
          href="/url/url/"
        >
          Вход
        </a>
      </div>
      <div
        class="css-bw50an"
      >
        <a
          class="css-som9qd"
          href="/url/url/"
        >
          Регистрация
        </a>
      </div>
    </div>
    <input
      class="css-1wjmxg1"
      name="email"
      placeholder="Введите логин или почту"
      required=""
      type="text"
    />
    <input
      class="css-1wjmxg1"
      name="code"
      placeholder="Введите проверочный код"
      required=""
      type="text"
    />
    <input
      class="css-1wjmxg1"
      id="password-input"
      name="password"
      placeholder="Введите новый пароль"
      required=""
      type="password"
    />
    <button
      class="css-19ic31z"
      type="submit"
    >
      Сбросить
    </button>
  </form>
</div>
`;

exports[`testing ResetPassword render Test clicks ResetPassword  3`] = `
<div>
  <form
    class="css-1jmybar"
  >
    <img
      class="css-1kpxy3o"
      src="logo.png"
    />
    <h1
      class="css-9jhoxj"
    >
      Сбросить пароль
    </h1>
    <div
      class="css-1bej84q"
    >
      <div
        class="css-bpveru"
      >
        <a
          class="css-som9qd"
          href="/url/url/"
        >
          Вход
        </a>
      </div>
      <div
        class="css-bw50an"
      >
        <a
          class="css-som9qd"
          href="/url/url/"
        >
          Регистрация
        </a>
      </div>
    </div>
    <input
      class="css-1wjmxg1"
      name="email"
      placeholder="Введите логин или почту"
      required=""
      type="text"
    />
    <input
      class="css-1wjmxg1"
      name="code"
      placeholder="Введите проверочный код"
      required=""
      type="text"
    />
    <input
      class="css-1wjmxg1"
      id="password-input"
      name="password"
      placeholder="Введите новый пароль"
      required=""
      type="password"
    />
    <button
      class="css-19ic31z"
      type="submit"
    >
      Сбросить
    </button>
  </form>
</div>
`;
